### PR TITLE
Add thunk, svg and dataset suites

### PR DESCRIPTION
**Pull request by buntec**

### CODE REVIEW

File: snabbdom/src/main/scala/snabbdom/modules/Dataset.scala [hunk 1]
```diff
@@ -71,7 +73,7 @@ object Dataset {
       oldDataset.foreach { case (key, _) =>
         dataset.get(key) match {
           case None =>
-            if (d != null) { // TODO: does this make sense?
+            if (js.isUndefined(d)) { // TODO: does this make sense?
```
Comment (armanbilge) [inline snabbdom/src/main/scala/snabbdom/modules/Dataset.scala:76]: It's here for now, we can revisit it in the future. At best it's only a micro-optimization anyway.

**Review by armanbilge — APPROVED** [review]
LGTM 🚀

File: snabbdom/src/main/scala/snabbdom/modules/Dataset.scala (hunk 1 above)
Comment (buntec) [reply snabbdom/src/main/scala/snabbdom/modules/Dataset.scala:75]: I think it should be `!js.isUndefined(d)`.

Comment (armanbilge) [reply snabbdom/src/main/scala/snabbdom/modules/Dataset.scala:75]: Wooops!

Comment (armanbilge) [reply snabbdom/src/main/scala/snabbdom/modules/Dataset.scala:75]:
```suggestion
            if (!js.isUndefined(d)) { // TODO: does this make sense?
```

File: snabbdom/src/main/scala/snabbdom/modules/Dataset.scala [hunk 2]
```diff
@@ -84,7 +85,7 @@ object Dataset {
 
       dataset.foreach { case (key, value) =>
         if (oldDataset.get(key).forall(_ != value)) {
-          if (d != null) { // TODO: does this make sense?
+          if (js.isUndefined(d)) { // TODO: does this make sense?
```
Comment (armanbilge) [inline snabbdom/src/main/scala/snabbdom/modules/Dataset.scala:88]:
```suggestion
          if (!js.isUndefined(d)) { // TODO: does this make sense?
```

**Pull request merged**